Lib/pathlib.py

```diff
@@ -421,6 +260,54 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _parse_parts(cls, parts):
+        parsed = []
+        sep = cls._flavour.sep
+        altsep = cls._flavour.altsep
+        drv = root = ''
+        it = reversed(parts)
+        for part in it:
+            if not part:
+                continue
+            if altsep:
+                part = part.replace(altsep, sep)
+            drv, rel = cls._flavour.splitdrive(part)
+            # According to POSIX path resolution:
+            # http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap04.html#tag_04_11
+            # "A pathname that begins with two successive slashes may be
+            # interpreted in an implementation-defined manner, although more
+            # than two leading slashes shall be treated as a single slash".
+            if cls._flavour is posixpath and rel[:2] == sep * 2 and rel[2:3] != sep:
+                root, rel = sep * 2, rel[2:]
+            elif rel[:1] == sep or drv[:1] == sep:
+                root, rel = sep, rel.lstrip(sep)
+            if sep in rel:
```

Comment:
If the fall-through is on purpose then a newline helps make that a bit more obvious.

```suggestion

            if sep in rel:
```